src/window.rs

```diff
@@ -90,16 +98,58 @@ impl Builder {
         };
 
         let builder = builder
-            .clone()
             .with_dimensions(self.dimensions.0, self.dimensions.1)
             .with_title(self.title.clone());
 
         let context = glutin::ContextBuilder::new()
             .with_vsync(self.vsync)
             .with_multisampling(self.multisampling);
 
+        let mut source_set = render::source::Set::default();
+        if let Some(path) = self.shader_directory.as_ref() {
+            let path = path.to_str().unwrap();
+            macro_rules! try_override {
+                ($name:ident) => {
+                    if let Ok(src) = render::Source::user(
```

Comment:
nit: let's use `match` against `if let ... else` here and below

Reply:
Much better idea.